Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -418,426 +465,36 @@ public class WhisperKit: Transcriber {
 
     // MARK: - Transcribe audio samples
 
-    public func transcribe(audioArray: [Float],
-                           decodeOptions: DecodingOptions? = nil,
-                           callback: TranscriptionCallback = nil) async throws -> TranscriptionResult?
-    {
-        progress.completedUnitCount = 0
-        if currentTimings == nil {
-            currentTimings = TranscriptionTimings()
-        }
-
-        if self.modelState != .loaded {
-            try await loadModels()
-        }
-
-        var timings = currentTimings!
-        timings.pipelineStart = CFAbsoluteTimeGetCurrent()
-
-        var options = decodeOptions ?? DecodingOptions()
-        options.verbose = Logging.shared.logLevel != .none
-        
-        var detectedLanguage: String?
-
-        let contentFrames = audioArray.count
-        timings.inputAudioSeconds = Double(Int(contentFrames) / WhisperKit.sampleRate) - Double(decodeOptions?.clipTimestamps.first ?? 0)
-
-        // MARK: Init decoder inputs
-
-        // These accumulate across windows
-        var allSegments: [TranscriptionSegment] = []
-        var allTokens: [Int] = []
-        var transcription = ""
-
-        guard let tokenizer = tokenizer else {
+    public func transcribe(
+        audioArray: [Float],
+        decodeOptions: DecodingOptions? = nil,
+        callback: TranscriptionCallback = nil
+    ) async throws -> [TranscriptionResult] {
```

Comment:
Breaking change, however in this case lets use a convenience method just above that has the same interface as before, and simply adjust the inputs and outputs to fit. This should be marked with a deprecated flag "Subject to removal in a future version, use {method signature} instead."